core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -159,47 +161,17 @@ public Schema createSchema(NameIdentifier ident, String comment, Map<String, Str
    */
   @Override
   public Schema loadSchema(NameIdentifier ident) throws NoSuchSchemaException {
-    NameIdentifier catalogIdentifier = getCatalogIdentifier(ident);
-    Schema schema =
-        doWithCatalog(
-            catalogIdentifier,
-            c -> c.doWithSchemaOps(s -> s.loadSchema(ident)),
-            NoSuchSchemaException.class);
+    EntityCombinedSchema schema =
+        TreeLockUtils.doWithTreeLock(ident, LockType.READ, () -> loadCombinedSchema(ident));
 
-    // If the Schema is maintained by the Gravitino's store, we don't have to load again.
-    boolean isManagedSchema = isManagedEntity(catalogIdentifier, Capability.Scope.SCHEMA);
-    if (isManagedSchema) {
-      return EntityCombinedSchema.of(schema)
-          .withHiddenPropertiesSet(
-              getHiddenPropertyNames(
-                  catalogIdentifier,
-                  HasPropertyMetadata::schemaPropertiesMetadata,
-                  schema.properties()));
+    if (schema == null || schema.imported()) {
```

Comment:
You should throw a `NoSuchSchemaException` instead of return null.

Reply:
Do you really think that method will return null, can you please carefully check the code?

Reply:
In the `exists` method, it will assume that the entity will return null. I keep the same behaviour with it.

Reply:
I checked the method won't return null. I will modify here. I just obey the same style as the method `exists`.

Reply:
I ever thought that we assume that this method may return null, I add the check of null.
```
  default boolean tableExists(NameIdentifier ident) {
    try {
      return loadTable(ident) != null;
    } catch (NoSuchTableException e) {
      return false;
    }
  }
```